arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -787,25 +789,29 @@ Result RocksDBCollection::truncateWithRangeDelete(transaction::Methods& trx) {
   if (!s.ok()) {
     return rocksutils::convertStatus(s);
   }
+  // we need crash server if any failure in order to trigger recovery,
+  // as rocksdb truncate already committed here
+  [&]() noexcept {
+    // post commit sequence
+    rocksdb::SequenceNumber seq = db->GetLatestSequenceNumber() - 1;
+
+    uint64_t numDocs = _meta.numberDocuments();
+    _meta.adjustNumberDocuments(seq,
```

Comment:
`RocksDBMetadata::adjustNumberDocuments()` can throw an exception in an emplace operation goes wrong (e.g. when there is no memory available).
In this case, the exception will bubble up from there into this function, which is marked `noexcept`. This will make the server crash with `std::terminate`.
If this behavior is intended, it is all fine, but it is not clear to me rn.

Reply:
Yes it's intentionally because if something throw after we put data in rocksdb we will state in the inconsistent state.

Reply:
Thanks, I already forgot about it.